cms/static/js/factories/import.js

```diff
@@ -1,30 +1,49 @@
 define([
-    'js/views/import', 'jquery', 'gettext', 'jquery.fileupload', 'jquery.cookie'
-], function(Import, $, gettext) {
+    'domReady', 'js/views/import', 'jquery', 'gettext', 'jquery.fileupload', 'jquery.cookie'
+], function(domReady, Import, $, gettext) {
+
     'use strict';
+
     return function (feedbackUrl, library) {
         var dbError;
+
         if (library) {
             dbError = gettext('There was an error while importing the new library to our database.');
         } else {
             dbError = gettext('There was an error while importing the new course to our database.');
         }
+
         var bar = $('.progress-bar'),
             fill = $('.progress-fill'),
             submitBtn = $('.submit-button'),
-            chooseBtn = $('.choose-file-button'),
+            chooseBtn = $('.view-import .choose-file-button'),
             defaults = [
                 gettext('There was an error during the upload process.') + '\n',
                 gettext('There was an error while unpacking the file.') + '\n',
                 gettext('There was an error while verifying the file you submitted.') + '\n',
                 dbError + '\n'
             ],
-            // Display the status of last file upload on page load
-            lastFileUpload = $.cookie('lastfileupload'),
+            previousImport = Import.storedImport(),
             file;
 
-        if (lastFileUpload){
-            Import.getAndStartUploadFeedback(feedbackUrl.replace('fillerName', lastFileUpload), lastFileUpload);
+        Import.callbacks.complete = function () {
```

Comment:
Just to be clear, something like:

```javascript
// factories/import.js

// to avoid code duplication
function onComplete() {
  bar.hide();
  chooseBtn
      .find('.copy').html(gettext("Choose new file")).end()
      .show();
}

return function (feedbackUrl, library) {
    // ...

    if (previousImport) {
        // ...
        Import.resume().then(onComplete);
    }

    $('#fileupload').fileupload({
        // ...
        Import.start(
            file.name,
            feedbackUrl.replace('fillerName', file.name)
       ).then(onComplete);
```